tst/functions/Mock.Tests.ps1

```diff
@@ -2779,3 +2779,51 @@ Describe "Debugging mocks" {
         }
     }
 }
+
+Describe "When inherited variables conflicts with parameters" {
+    BeforeAll {
+        Mock FunctionUnderTest { 'default' }
+        Mock FunctionUnderTest { 'filtered' } -ParameterFilter { $param1 -eq 'abc' } -Verifiable
+    }
+
+    It "parameterized mock should not be called due to inherited variable" {
+        $param1 = 'abc'
+        FunctionUnderTest | Should -Be 'default'
+    }
+
+    It "InvokeVerifiable should not pass due to test variable" {
+        # Uses same logic as mock execution, so should not be tricked
+        $param1 = 'abc'
+        FunctionUnderTest | Should -Be 'default'
+        { Should -InvokeVerifiable } | Should -Throw
+    }
+
+    It "KNOWN ISSUE: Should Invoke ParameterFilter will count false positive" {
+        # https://github.com/pester/Pester/issues/1873
+        FunctionUnderTest | Should -Be 'default'
+        FunctionUnderTest -param1 'abc' | Should -Be 'filtered'
+        $param1 = 'abc'
+
+        # This should show warning about conflict when in Diagnostic output (Mock debug message)
+        Should -Invoke FunctionUnderTest -ParameterFilter { $param1 -eq 'abc' } -Times 2 -Exactly
+    }
```

Comment:
```suggestion
    It "Should Invoke ParameterFilter will count false positive for the first FunctionUnderTest call" {
        # https://github.com/pester/Pester/issues/1873
        # this will pass the parameter filter because we define a variable param1 with the same name and value as the expected parameter value
        FunctionUnderTest | Should -Be 'default'
        FunctionUnderTest -param1 'abc' | Should -Be 'filtered'
        $param1 = 'abc'

        # This should show warning about conflict when in Diagnostic output (Mock debug message)
        # about already having a variable in the scope that is the same as parameter name
        Should -Invoke FunctionUnderTest -ParameterFilter { $param1 -eq 'abc' } -Times 2 -Exactly
    }
```

Reply:
That does not highlight the changes. I removed the Known issue as it sounds like we are doing something wrong that we don't expect. And added more details to comments. I did not change other code.